include/nbl/builtin/hlsl/bxdf/reflection/oren_nayar.hlsl

```diff
@@ -16,96 +18,37 @@ namespace bxdf
+    NBL_BXDF_CONFIG_ALIAS(quotient_pdf_type, Config);
+
+    NBL_CONSTEXPR_STATIC_INLINE BxDFClampMode _clamp = BxDFClampMode::BCM_MAX;
+
+    struct SCreationParams
     {
-        this_t retval;
-        retval._sample = _sample;
-        retval.interaction = interaction;
-        retval._clamp = _clamp;
-        return retval;
-    }
-
-    // iso
-    Scalar getNdotV() NBL_CONST_MEMBER_FUNC { return hlsl::mix(math::conditionalAbsOrMax<Scalar>(_clamp == BxDFClampMode::BCM_ABS, interaction.getNdotV(), 0.0), interaction.getNdotV(), _clamp == BxDFClampMode::BCM_NONE); }
-    Scalar getNdotVUnclamped() NBL_CONST_MEMBER_FUNC { return interaction.getNdotV(); }
-    Scalar getNdotV2() NBL_CONST_MEMBER_FUNC { return interaction.getNdotV2(); }
-    Scalar getNdotL() NBL_CONST_MEMBER_FUNC { return hlsl::mix(math::conditionalAbsOrMax<Scalar>(_clamp == BxDFClampMode::BCM_ABS, _sample.getNdotL(), 0.0), _sample.getNdotL(), _clamp == BxDFClampMode::BCM_NONE); }
-    Scalar getNdotLUnclamped() NBL_CONST_MEMBER_FUNC { return _sample.getNdotL(); }
-    Scalar getNdotL2() NBL_CONST_MEMBER_FUNC { return _sample.getNdotL2(); }
-    Scalar getVdotL() NBL_CONST_MEMBER_FUNC { return _sample.getVdotL(); }
-
-    // aniso
-    Scalar getTdotL2() NBL_CONST_MEMBER_FUNC { return _sample.getTdotL() * _sample.getTdotL(); }
-    Scalar getBdotL2() NBL_CONST_MEMBER_FUNC { return _sample.getBdotL() * _sample.getBdotL(); }
-    Scalar getTdotV2() NBL_CONST_MEMBER_FUNC { return interaction.getTdotV() * interaction.getTdotV(); }
-    Scalar getBdotV2() NBL_CONST_MEMBER_FUNC { return interaction.getBdotV() * interaction.getBdotV(); }
-
-    LS _sample;
-    SI interaction;
-    BxDFClampMode _clamp;
-};
-
-template<class LS, class Iso, class Aniso, class Spectrum NBL_PRIMARY_REQUIRES(LightSample<LS> && surface_interactions::Isotropic<Iso> && surface_interactions::Anisotropic<Aniso>)
-struct SOrenNayarBxDF
-{
-    using this_t = SOrenNayarBxDF<LS, Iso, Aniso, Spectrum>;
-    using scalar_type = typename LS::scalar_type;
-    using vector2_type = vector<scalar_type, 2>;
-    using ray_dir_info_type = typename LS::ray_dir_info_type;
-
-    using isotropic_interaction_type = Iso;
-    using anisotropic_interaction_type = Aniso;
-    using sample_type = LS;
-    using spectral_type = Spectrum;
-    using quotient_pdf_type = sampling::quotient_and_pdf<spectral_type, scalar_type>;
-    
-    using params_isotropic_t = OrenNayarParams<LS, Iso, scalar_type>;
-    using params_anisotropic_t = OrenNayarParams<LS, Aniso, scalar_type>;
-
+        scalar_type A;
+    };
+    using creation_type = SCreationParams;
 
     static this_t create(scalar_type A)
     {
         this_t retval;
         retval.A = A;
         return retval;
     }
+    static this_t create(NBL_CONST_REF_ARG(creation_type) params)
+    {
+        return create(params.A);
+    }
 
     scalar_type __rec_pi_factored_out_wo_clamps(scalar_type VdotL, scalar_type maxNdotL, scalar_type maxNdotV)
     {
```

Comment:
A2 and AB shall be a precomputed member too!

The idea is that if I want to save registers and spend ALU I do
```cpp
SOrenNayarBxDF<Config>::create(A).eval(_sample,interaction)
```
and if I want to spend registers and save ALU I do
```cpp
preCreatedOrenNayar.eval(_sample,interaction)
```